### PR TITLE
Filter cleanup and negation

### PR DESCRIPTION
**Pull request by winged**
This introduces filters that can be chained (same filter multiple times)
and negated.  Currently this is VERY WIP, but shows what can be done in
the allDocuments edge.

Code needs to be cleaned up and applied to the other filterable edges as well,
and tests adjusted etc..

### CODE REVIEW

**Comment by winged** [comment]
Example of what will be possible with this:
```graphql
{
  allDocuments(
   filter: [
    {
      searchAnswers: {
        questions: ["some-text"],
        value: "stuff that I want", lookup: TEXT
      }
    },
    {
      searchAnswers: {
        questions: ["some-text"],
        value: "things that I don't want", lookup: TEXT
      },
      invert:true
    }
  ]
  ) {
    edges {
      node {
        id
      }
    }
  }
}
```

**Comment by winged** [comment]
To easen the transition, I had the idea this weekend to provide the "old-style" filters as well for a release cycle or two. So instead of moving all the filters into the new structure, we would provde both interfaces. Shouldn't be too hard.

What do you guys think? @open-dynaMIX @anehx @kaldras @czosel

**Comment by czosel** [comment]
I really like the idea! Since we have a first release now, it would be good to get used to a workflow that involves deprecations and control the pace of breaking changes. Ideally, we could mark the old filters as deprecated directly in the schema.

**Comment by open-dynaMIX** [comment]
Great idea! :+1:

**Comment by winged** [comment]
Added the "old-style" filters back now. However even though I tried almost all day, I couldn't figure out how to mark them as deprecated. I'm tempted to get this merged nonetheless and just doing it in the documentation instead.

File: docs/graphql.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# GraphQL
+
+As noteds before, Caluma uses a GraphQL API for communication between
```
Comment (open-dynaMIX) [inline docs/graphql.md]:
```suggestion
As noted before, Caluma uses a GraphQL API for communication between
```

**Pull request merged**